docs/BehaviourInDetail.md

```diff
@@ -0,0 +1,93 @@
+# Behaviour in Detail
+## Nullness
+### Behaviour
+Nulls can always be produced for a field, except when a field is explicitly not null. 
+
+### Misleading Examples
+|Field is           |Null produced|
+|:------------------|:-----------:|
+|Of type X          | ✔ |
+|Not of type X      | ✔ |
+|In set [X, Y, ...] | ✔ |
+|Equal to X         | ✔ |
+|Greater than X     | ✔ |
+|Null               | ✔ |
+|Not null           | ❌ |
+
+Field A is equal to X and field B is equal to Y if field A is equal to X:
```

Comment:
This example could probably do with being a bit wordier to explain it

Reply:
Would a profile snippet be useful here?

Reply:
Or using the following wording?
_Consider a profile where field A is equal to X and field B is equal to Y if field A is equal to X.
This effectively is an if statement but with the "if" condition forced to be true._
Or another wording?